### PR TITLE
feat: extract intensity components

### LINKED ISSUE #184: Extract Intensity components from total intensity
This is required when calculating fit fractions and visualizing different components of the full intensity.

### PR DESCRIPTION
**Pull request by redeboer**
Closes #184

### CODE REVIEW

**Comment by redeboer** [comment]
@spflueger Currently, the functions are only implemented in usage/step3 notebook. We have to think where and how this can be integrated into the package.

File: src/tensorwaves/model.py [hunk 1]
```diff
@@ -109,13 +119,6 @@ def __init__(
         if not all(map(lambda p: isinstance(p, sp.Symbol), parameters)):
             raise TypeError(f"Not all parameters are of type {sp.Symbol}")
 
-        if not set(parameters) <= set(expression.free_symbols):
```
Comment (spflueger) [inline src/tensorwaves/model.py:112]: why is this removed?

File: src/tensorwaves/model.py [hunk 2]
```diff
@@ -180,3 +183,49 @@ def variables(self) -> FrozenSet[str]:
     @property
     def argument_order(self) -> Tuple[str, ...]:
         return tuple(x.name for x in self.__argument_order)
+
+
+def add_components(
```
Comment (spflueger) [inline src/tensorwaves/model.py:188]: How about we put this into some new module, since this is quite special code related to the `HelicityModel`.

File: src/tensorwaves/model.py (hunk 1 above)
Comment (redeboer) [reply src/tensorwaves/model.py:112]: Because if you take a sub-expression that does not feature all the parameters that were in the original model, this will crash. I don't think a check like this is necessary here.

File: src/tensorwaves/model.py (hunk 2 above)
Comment (redeboer) [reply src/tensorwaves/model.py:188]: Yes, I suppose we need to re-introduce the `physics` module (but now separate from the `model` module). #15, #14, and #38 could be put there as well.

File: src/tensorwaves/model.py (hunk 1 above)
Comment (redeboer) [reply src/tensorwaves/model.py:112]: Possible, but then the caller always have to filter those beforehand useing `free_symbols`.

File: src/tensorwaves/model.py (hunk 2 above)
Comment (spflueger) [reply src/tensorwaves/model.py:188]: #15 and #38 belong together (model domain related)

#14 not related to these, but related to this PR. So we could def put those together. Maybe into `fit_fractions.py` or `components.py`.

I simply do not want `model.py` to contain model specific implementations.

File: src/tensorwaves/model.py (hunk 1 above)
Comment (redeboer) [reply src/tensorwaves/model.py:112]: Added the chack back but as warning e08f084

File: src/tensorwaves/model.py (hunk 2 above)
Comment (redeboer) [reply src/tensorwaves/model.py]: Ok, moved them to a separate (possibly temporary) `physics` module (b2a190d). The location of these functions will become clearer in through those other issues
